python/lsst/fgcmcal/fgcmBuildStarsBase.py

```diff
@@ -518,8 +531,14 @@ def _makeSourceMapper(self, sourceSchema):
         sourceMapper.addMapping(sourceSchema['coord_dec'].asKey(), 'dec')
         sourceMapper.addMapping(sourceSchema['slot_Centroid_x'].asKey(), 'x')
         sourceMapper.addMapping(sourceSchema['slot_Centroid_y'].asKey(), 'y')
-        sourceMapper.addMapping(sourceSchema[self.config.psfCandidateName].asKey(),
-                                'psf_candidate')
+        try:
+            sourceMapper.addMapping(sourceSchema[self.config.psfCandidateName].asKey(),
+                                    'psf_candidate')
+        except LookupError:
+            sourceMapper.editOutputSchema().addField(
+                "psf_candidate", type='Flag',
+                doc=("Flag set if the source was a candidate for PSF determination, "
+                     "as determined by the star selector."))
```

Comment:
The mapping maps the input to the output catalog; if it's not in the input catalog, then it just will be all `False` in the output catalog, which is fine.  And the field would be missing in older catalogs (such as the old test data!).  I didn't want to require it since it isn't essential for the main operations.